sematic/resolvers/local_resolver.py

```diff
@@ -578,6 +655,67 @@ def _save_graph(self):
         self._buffer_artifacts.clear()
         self._buffer_edges.clear()
 
+    def _get_artifact_from_cache(self, run: Run) -> Optional[Artifact]:
+        """
+        Attempt to get the Artifact for the specified Run from the cache.
+
+        Parameters
+        ----------
+        run: Run
+            The Run for which to look up the Artifact.
+
+        Returns
+        -------
+        The Artifact which resolves the Run, or None if none was found.
+        """
+        if run.cache_key is None:
+            return None
+        
+        # TODO: always get the original run, not any run that matches
+        # TODO: check cache key collision
```

Comment:
These should be implemented in this PR / before this PR in this PR chain.
A direct consequence is that once a cached `func`'s execution fails during development, subsequent runs can get either a successfully resolved or a failed hit here, causing the run to be randomly executed or not.

Reply:
This is resolved by always taking the first successful result.